imap/jmap_contact.c

```diff
@@ -1549,180 +1549,157 @@ static json_t *jmap_contact_from_vcard(const char *userid,
     json_object_set_new(obj, "jobTitle", jmap_utf8string(item));
 
     json_t *adr = json_array();
+    json_t *emails = json_array();
+    json_t *phones = json_array();
+    json_t *online = json_array();
+    int emailIndex = 0, defaultEmailIndex = -1;
 
     for (entry = card->properties; entry; entry = entry->next) {
-        if (strcasecmp(entry->name, "adr")) continue;
-        json_t *item = json_object();
-
-        /* XXX - type and label */
-        const strarray_t *a = entry->v.values;
-
         const struct vparse_param *param;
-        const char *type = "other";
         const char *label = NULL;
+        size_t label_len = 0;
+
+        /* Apple label */
         if (entry->group) {
             label = hash_lookup(entry->group, &labels);
+            label_len = strlen(label);
+
+            /* Check and adjust for weird (localized?) labels */
+            if (label_len > 8 && !strncmp(label, "_$!<", 4)) {
```

Comment:
I don't know how these weird labels look like (is there a test for it?). But the pointer arithmetic here looks confusing. Why is label increment by 4, but its length decrement by 8? Is there some hidden assumption?

Alternatively: couldn't the code just skip any non-alphanumeric characters at the start of a label?

Reply:
The Apple labels look something like _$!<Foo>!$_

Reply:
So should we just skip special chars at the start of the label? And decrement label_len by the count of skipped characters?

Reply:
Damn it, GitHub tried to translate the special chars that I typed.  The label starts with _$!< and ends with >!$_, which is where the 8 comes from

Reply:
GitHub still stripping the underscores

Reply:
Today I learned that Github does some funky stuff when displaying code snippets.

Reply:
If you're commenting in the web interface, you can make it format it weird stuff properly (skipping the automatic markdown italics/bold/etc) by wrapping it in backticks -- so `_$!<` and `>!$_` -- and you can check the Preview tab before submitting to make sure it understands you correctly.

But if you're commenting by replying to the notification email, it's anyone's guess what it'll do.  I think backticks will still work the same way, in the same way as _wrapping something in underscores italicises it_ etc.  But without the Preview button, you don't get an opportunity to check that it understood you correctly before you commit to posting it like that.